src/components/SwapEtherspotKlimaZap.tsx

```diff
@@ -0,0 +1,1189 @@
+      )
+    }
+    if (!web3.account) {
+      return <ConnectButton size="large" />
+    }
+    if (fromChainKey && web3.chainId !== getChainByKey(fromChainKey).id) {
+      const fromChain = getChainByKey(fromChainKey)
+      return (
+        <Button
+          shape="round"
+          type="primary"
+          icon={<SwapOutlined />}
+          size={'large'}
+          htmlType="submit"
+          onClick={() => switchChain(fromChain.id)}>
+          Switch Network to {fromChain.name}
+        </Button>
+      )
+    }
+    if (routesLoading) {
+      return (
+        <Button
+          disabled={true}
+          shape="round"
+          type="primary"
+          icon={<SyncOutlined spin />}
+          size={'large'}>
+          Searching Routes...
+        </Button>
+      )
+    }
+    if (noRoutesAvailable) {
+      return (
+        <Button disabled={true} shape="round" type="primary" size={'large'}>
+          No Route Found
+        </Button>
+      )
+    }
+    if (!hasSufficientGasBalanceOnStartChain(route.lifiRoute)) {
+      return (
+        <Button disabled={true} shape="round" type="primary" size={'large'}>
+          Insufficient Gas on Start Chain
+        </Button>
+      )
+    }
+    if (!hasSufficientGasBalanceOnCrossChain(route.lifiRoute)) {
+      return (
+        <Tooltip title="The selected route requires a swap on the chain you are tranferring to. You need to have gas on that chain to pay for the transaction there.">
+          <Button disabled={true} shape="round" type="primary" size={'large'}>
+            Insufficient Gas on Destination Chain
+          </Button>
+        </Tooltip>
+      )
+    }
+    if (!hasSufficientBalance()) {
+      return (
+        <Button disabled={true} shape="round" type="primary" size={'large'}>
+          Insufficient Funds
+        </Button>
+      )
+    }
+
+    return (
+      <Button
+        disabled={highlightedIndex === -1}
+        shape="round"
+        type="primary"
+        size={'large'}
+        onClick={() => openModal()}>
+        Stake
+      </Button>
+    )
+  }
+
+  const toSection = () => {
+    const amount = route?.klimaStep?.estimate?.toAmountMin || '0'
+    const formattedAmount = tokenPolygonSKLIMA ? formatTokenAmount(tokenPolygonSKLIMA, amount) : '0'
+    return (
+      <Row
+        style={{
+          marginTop: '32px',
+        }}
+        gutter={[
+          { xs: 8, sm: 16 },
+          { xs: 8, sm: 16 },
+        ]}>
+        <Col span={10}>
+          <div className="form-text">To:</div>
+        </Col>
+        <Col span={14}>
+          <div className="form-input-wrapper">
+            <Input
+              type="text"
+              value={`${formattedAmount}`}
+              bordered={false}
+              disabled
+              style={{ color: 'rgba(0, 0, 0, 0.85)', fontWeight: '400' }}
+            />
+          </div>
+        </Col>
+      </Row>
+    )
+  }
+
+  const isTransferto = window.location.href.includes('transferto')
```

Comment:
I guess we don't need those checks anymore, maybe we should remove them as part of a separate story?

Reply:
Seperate story! This PR is already big enough!